includes/gateways/stripe/includes/class-give-stripe-customer.php

```diff
@@ -410,7 +411,7 @@ public function attach_source() {
 			// Fetch the new card or source object to match with customer attached card fingerprint.
 			if ( give_stripe_is_source_type( $this->payment_method_id, 'tok' ) ) {
 				$token_details = $this->stripe_gateway->get_token_details( $this->payment_method_id );
-				$new_card      = $token_details->card;
+				$new_card      = $token_details->bank_account;
```

Comment:
@ravinderk why was this changed from `$token_details->card` to `$token_details->bank_account`? The return value is being stored as the variable `$new_card`, so I would expect to see a reference to `card` and not `bank_account`.

Reply:
I see in the PR description that `card` is not defined, but we are still setting the value as `$new_card`, so I'm confused.

Reply:
@kjohnson this whole logic needs to refactor. I tried to touch code which was causing PHP notice. `$new_card` is equivalent to a new source. I renamed it to make it understandable: https://github.com/impress-org/givewp/commit/cd3a64a9bb46da6a0fb3a9badd658a9d4546867d

Stripe Source object: https://stripe.com/docs/api/sources/object